cmd/notary/integration_test.go

```diff
@@ -1366,3 +1372,183 @@ func TestPurge(t *testing.T) {
+	privKey, err := utils.GenerateECDSAKey(rand.Reader)
+	require.NoError(t, err)
+
+	pemBytes, err := utils.EncryptPrivateKey(privKey, "root", "", "")
+	require.NoError(t, err)
+
+	nBytes, err := tempFile.Write(pemBytes)
+	require.NoError(t, err)
+	tempFile.Close()
+	require.Equal(t, len(pemBytes), nBytes)
+	rootKeyID = privKey.ID()
+
+	// import the key
+	_, err = runCommand(t, tempDir, "key", "import", tempFile.Name())
+	require.NoError(t, err)
+
+	// if there is hardware available, root will only be on hardware, and not
+	// on disk
+	newRoot, _ := assertNumKeys(t, tempDir, 1, 0, !rootOnHardware())
+	require.Equal(t, rootKeyID, newRoot[0])
+
+	// test 2, no path but role flag included with unencrypted key
+
+	tempFile2, err := ioutil.TempFile("", "pemfile")
+	require.NoError(t, err)
+	// close later, because we might need to write to it
+	defer os.Remove(tempFile2.Name())
+
+	privKey, err = utils.GenerateECDSAKey(rand.Reader)
+	require.NoError(t, err)
+
+	pemBytes, err = utils.EncryptPrivateKey(privKey, "", "", "")
+	require.NoError(t, err)
+
+	nBytes, err = tempFile2.Write(pemBytes)
+	require.NoError(t, err)
+	tempFile2.Close()
+	require.Equal(t, len(pemBytes), nBytes)
+
+	// import the key
+	_, err = runCommand(t, tempDir, "key", "import", tempFile2.Name(), "-r", "root")
+	require.NoError(t, err)
+
+	// if there is hardware available, root will only be on hardware, and not
+	// on disk
+	_, _ = assertNumKeys(t, tempDir, 2, 0, !rootOnHardware())
+
+	// test 3, no path no role included with unencrypted key
+
+	tempFile3, err := ioutil.TempFile("", "pemfile")
+	require.NoError(t, err)
+	// close later, because we might need to write to it
+	defer os.Remove(tempFile3.Name())
+
+	privKey, err = utils.GenerateECDSAKey(rand.Reader)
+	require.NoError(t, err)
+
+	pemBytes, err = utils.EncryptPrivateKey(privKey, "", "", "")
+	require.NoError(t, err)
+
+	nBytes, err = tempFile3.Write(pemBytes)
+	require.NoError(t, err)
+	tempFile2.Close()
+	require.Equal(t, len(pemBytes), nBytes)
+
+	// import the key
+	_, err = runCommand(t, tempDir, "key", "import", tempFile3.Name())
+	require.NoError(t, err)
+
+	// if there is hardware available, root will only be on hardware, and not
+	// on disk
+	_, _ = assertNumKeys(t, tempDir, 2, 0, !rootOnHardware())
+
+	// test 4, no path non root role with no gun flag with unencrypted key
+
+	tempFile4, err := ioutil.TempFile("", "pemfile")
+	require.NoError(t, err)
+	// close later, because we might need to write to it
+	defer os.Remove(tempFile4.Name())
+
+	privKey, err = utils.GenerateECDSAKey(rand.Reader)
+	require.NoError(t, err)
+
+	pemBytes, err = utils.EncryptPrivateKey(privKey, "", "", "")
+	require.NoError(t, err)
+
+	nBytes, err = tempFile4.Write(pemBytes)
+	require.NoError(t, err)
+	tempFile4.Close()
+	require.Equal(t, len(pemBytes), nBytes)
+	newKeyID := privKey.ID()
+
+	// import the key
+	_, err = runCommand(t, tempDir, "key", "import", tempFile4.Name(), "-r", "somerole")
```

Comment:
I think the first two look correct.  For the second two, assuming that the key does not have a role specified:

> `notary key import my.key -r targets/releases -g randomgun: key` is imported to tuf_keys/ with role: notary.DefaultImportRole

Should this be imported with the provided role (`targets/releases`)?  Or are we overriding all delegation roles to just be "delegation"?

> `notary key import my.key -r snapshot -g randomgun`: key is imported to tuf_keys/randomgun/ with role: notary.DefaultImportRole

Should this be imported with the role "snapshot" since that was provided as a flag?